### ISSUE TITLE
How to incorporate oracledb with webdriver flow control mechanism?

### ISSUE THREAD

**Issue by garimamathur16** [oracle/node-oracledb#110]
Hi,
I am new to oracledb module. Would appreciate guidance on the same.

I need to execute a few database step as a part of my test case(protractor). In order to execute it in a specific order(in my case, after my record has been submitted successfully on UI, I have to check the record in database), I have used Webdrivejs control flow: flow.execute() method. 
That works fine with user-defined functions.
But I am facing issues with database connection creation(perhaps, due to the callback)

```
flow = protractor.promise.controlFlow();
flow.execute(function verify_db_record() {dbhelp.execute_query("**query**",'param')});

execute_query :function(query,queryparam)
	 {
	        oracledb.getConnection(connectData,function(err, connection)
		{if (err){console.error(err.message);return;}
		wait.launchFiber(runQuery,query,queryparam,connection);
	 }
function runQuery(query,queryparam,conn)
   {
	try {
	  var result = wait.forMethod(conn, "execute", query, [queryparam],{isAutoCommit:true}); 
	  console.log(result.rows);
	  if(result.rows)
	  {status=true }
	} 
	catch(err) {
	   console.log(err);
	}
	console.log("query end: ",query);
	expect(status).toBe(true);
  }
```

Please help me understand, if there is any alternative to run the query in a sequence.

Thanks in advance.

**Comment by cjbj** [comment]
What's the actual issue you are having?

Do you have a complete, stand-alone testcase you can share that we can try to run?

**Comment by garimamathur16** [comment]
Thanks for being prompt.

The issue I am facing is that, as per above example: when function execute_query is called, it does not returns connection.
flow.execute(function verify_db_record() {dbhelp.execute_query("**query**",'param')});
 As a result I get this error: (since, conn is undefined[connection not created])
[TypeError: Cannot read property 'execute' of undefined]

However, If I remove flow.execute() and call the function execute_query directly, my query is executed successfully, but asynchronously.

In layman language, steps for the test case would be:
a) Login into the application and create a record by submitting some data.
b) After Submit, verify in DB, the record is inserted correctly.
c) As a part of cleanup, delete the record inserted from database.

**Comment by cjbj** [comment]
@garimamathur16 I have an old note about the write only properties of a connection causing issues for promises.  If you have a standalone testcase you can give me, that would be really helpful.
